src/spawn.rs

```diff
@@ -0,0 +1,168 @@
+//! Entity spawning utilities.
+
+// Unused utilities may trigger these lints undesirably.
+#![allow(dead_code)]
+
+use bevy::ecs::system::EntityCommand;
+use bevy::ecs::system::EntityCommands;
+use bevy::ecs::system::RunSystemOnce as _;
+use bevy::prelude::*;
+
+/// Re-exported extension traits.
+#[allow(unused_imports)]
+pub mod prelude {
+    pub use super::{
+        AddExt as _, AddFnExt as _, SpawnExt as _, SpawnSystemExt as _, WorldSpawnExt as _,
+    };
+}
+
+/// An extension trait that provides helper functions for deferred entity spawning.
+pub trait SpawnExt {
+    // Workaround for https://github.com/bevyengine/bevy/issues/14231#issuecomment-2216321086.
+    /// Spawn an entity with an [`EntityCommand`].
+    fn spawn_with<M: 'static>(&mut self, command: impl EntityCommand<M>) -> EntityCommands;
+
+    /// Spawn an entity with a [`System`] that receives the new entity ID via [`In<Entity>`].
+    fn spawn_fn<M>(
```

Comment:
The distinction between `spawn_with` and `spawn_fn` is currently probably impossible to grasp from the naming alone. I suggest renaming this to `insert_with`.

Reply:
These both spawn a new entity, though. The difference is whether we're passing in data.

Reply:
Oh wait, you're right. Can a system implement `EntityCommand` so that we can remove this function?

Reply:
Due to the orphan rule, that would have to be done upstream. But it would be nice if so.

Reply:
Fair enough. We still need a better name though.

Reply:
Actually, in this case you *can* bypass the orphan rule by wrapping the system in a struct and then implementing traits on that struct. Not an option for the template because that is severe type magic, but just wanted to let you know.